api/src/iosMain/kotlin/com.giancarlobuenaflor.kflogger/util/NSThreadBasedCallerFinder.kt

```diff
@@ -0,0 +1,52 @@
+package com.giancarlobuenaflor.kflogger.util
+
+import com.giancarlobuenaflor.kflogger.KAbstractLogger
+import com.giancarlobuenaflor.kflogger.KLogSite
+import com.giancarlobuenaflor.kflogger.Klass
+import com.giancarlobuenaflor.kflogger.backend.KPlatformLogCallerFinder
+import platform.Foundation.NSThread
+
+public class NSThreadBasedCallerFinder private constructor() : KPlatformLogCallerFinder() {
+  private fun findCallerNameOf(target: Klass<*>): String {
+    val name = target.kClass.qualifiedName
+    checkNotNull(name) { "Logger class has no name" }
+    val callstackSymbols = NSThread.callStackSymbols
+    check(callstackSymbols.size > 1) { "Callstack is too short" }
+    val firstIndex =
+        findIndexOfStringContainingSubstring(
+            callstackSymbols as List<String>, target.kClass.qualifiedName + ".")
+    val stackframeString = callstackSymbols[firstIndex + 1]
+    val stackFrameSignature = stackframeString.substringAfter("kfun:").substringBefore("#")
+    val lastIndexOfDot = stackFrameSignature.lastIndexOf(".")
+    check(lastIndexOfDot != -1) { "Stackframe signature does not contain a dot" }
+    if (stackFrameSignature[lastIndexOfDot + 1].isLowerCase()) {
+      // If the last character before the dot is lowercase, we assume it's a function name
+      return stackFrameSignature.substring(0, lastIndexOfDot)
+    }
+    return stackFrameSignature
+  }
+
+  private fun findIndexOfStringContainingSubstring(
```

Comment:
This function can be simplified by calling [`indexOfFirst`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/index-of-first.html):
```
stackTrace.indexOfFirst {
  it.contains(substring)
}
```

Once simplified it should just be inlined at the call site above